server/src/main/java/org/elasticsearch/rest/action/search/SearchResponseMetrics.java

```diff
@@ -8,31 +8,55 @@
 
 package org.elasticsearch.rest.action.search;
 
+import org.elasticsearch.telemetry.metric.LongCounter;
 import org.elasticsearch.telemetry.metric.LongHistogram;
 import org.elasticsearch.telemetry.metric.MeterRegistry;
 
+import java.util.Map;
+
 public class SearchResponseMetrics {
 
+    public enum ResponseCountTotalStatus {
+        SUCCESS,
+        PARTIAL_FAILURE,
+        FAILURE
+    }
+
+    public static final String RESPONSE_COUNT_TOTAL_STATUS_ATTRIBUTE_NAME = "status";
+
     public static final String TOOK_DURATION_TOTAL_HISTOGRAM_NAME = "es.search_response.took_durations.histogram";
+    public static final String RESPONSE_COUNT_TOTAL_COUNTER_NAME = "es.search_response.response_count.total";
 
     private final LongHistogram tookDurationTotalMillisHistogram;
+    private final LongCounter responseCountTotalCounter;
 
     public SearchResponseMetrics(MeterRegistry meterRegistry) {
         this(
             meterRegistry.registerLongHistogram(
                 TOOK_DURATION_TOTAL_HISTOGRAM_NAME,
                 "The SearchResponse.took durations in milliseconds, expressed as a histogram",
                 "millis"
+            ),
+            meterRegistry.registerLongCounter(
+                RESPONSE_COUNT_TOTAL_COUNTER_NAME,
+                "The cumulative total of search responses with an attribute to describe"
+                    + "success, partial failure, or failure, expressed as a counter and an attribute",
```

Comment:
nit: I think you need a space after "describe" or before "success"

Reply:
Asking to learn: what does "expressed as a counter and an attribute" mean? How is it both?